staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -193,20 +195,21 @@ type udsGRPCConnector struct {
 	udsName string
 }
 
-func (u *udsGRPCConnector) connect() (proxier, error) {
+func (u *udsGRPCConnector) connect(ctx context.Context) (proxier, error) {
 	udsName := u.udsName
-	dialOption := grpc.WithContextDialer(func(context.Context, string) (net.Conn, error) {
-		c, err := net.Dial("unix", udsName)
+	dialOption := grpc.WithContextDialer(func(ctx context.Context, path string) (net.Conn, error) {
+		dialer := &net.Dialer{}
+		c, err := dialer.DialContext(ctx, "unix", udsName)
 		if err != nil {
 			klog.Errorf("failed to create connection to uds name %s, error: %v", udsName, err)
+			return nil, fmt.Errorf("grpc-connect(uds): %w", err)
 		}
-		return c, err
+		return c, nil
 	})
 
-	ctx := context.TODO()
 	tunnel, err := client.CreateSingleUseGrpcTunnel(ctx, udsName, dialOption, grpc.WithInsecure())
```

Comment:
Discussed offline, the `tunnel` is reused by multiple requests, so passing in the ctx to CreateSingleUseGrpcTunnel causes one request's ctx affecting other requests.

Reply:
so CreateSingleUseGrpcTunnel ... isn't?